src/objecttypes/utils/views.py

```diff
@@ -23,3 +36,45 @@ def server_error(request, template_name=ERROR_500_TEMPLATE_NAME):
         )
     context = {"request": request}
     return http.HttpResponseServerError(template.render(context))
+
+
+def exception_handler(exc, context):
+    """
+    Transform 5xx errors into DSO-compliant shape.
+    """
+    response = drf_exception_handler(exc, context)
+    if not response:
+        if config("DEBUG", default=False):
+            return None
+
+        data = {
+            "code": "error",
+            "title": "Internal Server Error",
+            "status": status.HTTP_500_INTERNAL_SERVER_ERROR,
+            "detail": _("A server error has occurred."),
+        }
+        event = "api.uncaught_exception"
+
+        if isinstance(exc, DatabaseError) and "jsonpath" in exc.args[0]:
+            # provide user-friendly response if data_icontains was used but DB couldn't process it
+            data["detail"] = (
+                "This search operation is not supported by the underlying data store."
+            )
+            event = "api.database_exception"
```

Comment:
Objecttypes doesn't have the query param `data_icontains`, so let's leave this out

```suggestion
```